front/src/Weekview.js

```diff
@@ -113,7 +124,44 @@ class Weekview extends Component {
         var week1 = new Date(date1.getFullYear(), 0, 4);
         var current = 1 + Math.round(((date1.getTime() - week1.getTime()) / 86400000
         - 3 + (week1.getDay() + 6) % 7) / 7);
-        this.setState({weekNro: current})
+
+        //Tää asettaa sen mikä viikkonumero on alotusnäytöllä
+        //Nyt tarvis ottaa tähän url parametreistä se viikkonumero
+        //Jos ei parametrejä nii sit toi current. Muuten parametrien
+
+        //Urlista lasketaan oikee viikkonumero
+        try {
+            let fullUrl = window.location.href.split("/");
+            let urlParamDate = fullUrl[5];
+    
+            let urlParamDateSplit = urlParamDate.split("-")
+    
+            var weeknumber = moment(urlParamDate, "YYYYMMDD").week();
+            var daynumber = urlParamDate
+
+            let paramDay = urlParamDateSplit[2]
+            let paramMonth = urlParamDateSplit[1]
+            let paramYear = urlParamDateSplit[0]
+    
+            let paramDateCorrect = moment(paramYear + paramMonth + paramDay, "YYYYMMDD")
+    
+            //Jos viikkonumero ei oo oikee laitetaan current
+            if (isNaN(weeknumber)) {
+                this.setState({weekNro: current}) 
+                this.props.history.replace("/weekview/")
+            }
+            else {
+            //Jos on oikee nii laitetaan url params
+            //miksei date settaannu!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+            this.setState({weekNro: weeknumber, date: paramDateCorrect})
+            console.log(this.state.date);
+            }   
+        }
+        catch {
```

Comment:
I suggest catching the error too, so
catch (error) {
... code here
// maybe even console.log(error);
}